rest/config_test.go

```diff
@@ -799,3 +800,56 @@ func TestParseCommandLineWithConfigContent(t *testing.T) {
 	assert.False(t, guest.Disabled)
 	assert.Equal(t, base.SetFromArray([]string{"*"}), guest.ExplicitChannels)
 }
+
+// This test uses both test_data_bucket and test_indexbucket buckets; these buckets should be available
+// on Couchbase server instance before running this test.
+func TestValidateAndAddDBFromConfig(t *testing.T) {
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("ValidateAndAddDBFromConfig test require Couchbase Bucket")
+	}
+	tmpDir, err := ioutil.TempDir("", "sync_gateway")
```

Comment:
Wherever possible, we need to avoid touching the filesystem in tests. It's already caused too many problems, and makes the test way bigger than it needs to be.

Can't we initialize a `ServerConfig` struct directly in the test?

Reply:
Yeah, this makes sense. It is revised in latest revision.